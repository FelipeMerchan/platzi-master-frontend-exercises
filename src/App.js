import React from 'react'
import { BrowserRouter, Switch, Route } from 'react-router-dom'

import Home from './pages/Home'
import FirstWeek from './pages/FirstWeek'
import FourthWeek from './pages/FourthWeek'
import FifthWeek from './pages/FifthWeek'
import SixthWeek from './pages/SixthWeek'
import ThirteenthWeek from './pages/ThirteenthWeek'
import './globals.css'

function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path={`${process.env.PUBLIC_URL}/`} component={ Home } />
        <Route exact path={`${process.env.PUBLIC_URL}/primera-semana`} component={ FirstWeek } />
        <Route exact path={`${process.env.PUBLIC_URL}/cuarta-semana`} component={ FourthWeek } />
        <Route exact path={`${process.env.PUBLIC_URL}/quinta-semana`} component={ FifthWeek } />
        <Route exact path={`${process.env.PUBLIC_URL}/sexta-semana`} component={ SixthWeek } />
        <Route exact path={`${process.env.PUBLIC_URL}/decima-tercera-semana`} component={ ThirteenthWeek } />
      </Switch>
    </BrowserRouter>
  )
}

export default App;