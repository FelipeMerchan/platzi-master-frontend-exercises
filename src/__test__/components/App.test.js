import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import App from '../../App'

describe('<App />', () => {
  const app = shallow(<App />)

  test('Render del componente App',() => {
    expect(app.length).toEqual(1)
  })
})

describe('App Snapshot', () => {
  test('Comprobar la interfaz de usuario del componente App', () => {
    const app = create(<App />)
    expect(app.toJSON()).toMatchSnapshot()
  })
})