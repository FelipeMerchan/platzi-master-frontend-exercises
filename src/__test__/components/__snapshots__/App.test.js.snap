// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Comprobar la interfaz de usuario del componente App 1`] = `
<main
  className="u-wrapper bg-dark-blue"
>
  <div
    className="wrapper"
  >
    <header
      className="HeaderHome"
    >
      <img
        alt="Felipe Sc"
        className="HeaderHome__brand"
        src="felipe-sc-imagotipo.svg"
        width="220"
      />
    </header>
    <section
      className="Hero"
    >
      <figure
        className="Hero__imageContainer"
      >
        <img
          alt="Felipe Sc"
          className="Hero__image"
          src="felipe-sc-photo@2x.png"
          width="400"
        />
        <div />
      </figure>
      <h1
        className="Hero__title text-uppercase"
      >
        Ejercicios
        <br />
        <span>
          frontend
        </span>
      </h1>
      <h3
        className="Hero__program text-uppercase"
      >
        Platzi Master
      </h3>
      <a
        className="Hero__scroll-down text-uppercase"
        href="#exercises"
      >
        Scroll down
      </a>
    </section>
    <div>
      <section
        className="Exercises"
        id="exercises"
      >
        <h2
          className="Exercises__title text-uppercase"
        >
          Explora los ejercicios
        </h2>
        <div
          className="ExercisesLists"
        >
          <article
            className="Exercise"
          >
            <figure
              className="Exercise__imageContainer"
            >
              <img
                alt="Página del periódico The New York Times"
                className="Exercise__image"
                loading="lazy"
                src="primera-semana-poster.png"
              />
            </figure>
            <div
              className="Exercise__details d-flex"
            >
              <h3
                className="Exercise__title text-uppercase"
              >
                Semana: 1
              </h3>
              <h5
                className="Exercise__difficulty text-uppercase"
              >
                Fácil
              </h5>
              <a
                className="Exercise__link text-uppercase"
                exact="true"
                href="/primera-semana"
                onClick={[Function]}
              >
                Descubrir
              </a>
            </div>
            <div
              className="Exercise__border"
            />
          </article>
          <article
            className="Exercise"
          >
            <figure
              className="Exercise__imageContainer"
            >
              <img
                alt="Página del curso de HTMl y CSS práctico"
                className="Exercise__image"
                loading="lazy"
                src="cuarta-semana-poster.png"
              />
            </figure>
            <div
              className="Exercise__details d-flex"
            >
              <h3
                className="Exercise__title text-uppercase"
              >
                Semana: 4
              </h3>
              <h5
                className="Exercise__difficulty text-uppercase"
              >
                Fácil
              </h5>
              <a
                className="Exercise__link text-uppercase"
                exact="true"
                href="/cuarta-semana"
                onClick={[Function]}
              >
                Descubrir
              </a>
            </div>
            <div
              className="Exercise__border"
            />
          </article>
          <article
            className="Exercise"
          >
            <figure
              className="Exercise__imageContainer"
            >
              <img
                alt="Página de ejercicio web"
                className="Exercise__image"
                loading="lazy"
                src="quinta-semana-poster.png"
              />
            </figure>
            <div
              className="Exercise__details d-flex"
            >
              <h3
                className="Exercise__title text-uppercase"
              >
                Semana: 5
              </h3>
              <h5
                className="Exercise__difficulty text-uppercase"
              >
                Normal
              </h5>
              <a
                className="Exercise__link text-uppercase"
                exact="true"
                href="/quinta-semana"
                onClick={[Function]}
              >
                Descubrir
              </a>
            </div>
            <div
              className="Exercise__border"
            />
          </article>
          <article
            className="Exercise"
          >
            <figure
              className="Exercise__imageContainer"
            >
              <img
                alt="Página para empresa"
                className="Exercise__image"
                loading="lazy"
                src="sexta-semana-poster.png"
              />
            </figure>
            <div
              className="Exercise__details d-flex"
            >
              <h3
                className="Exercise__title text-uppercase"
              >
                Semana: 6
              </h3>
              <h5
                className="Exercise__difficulty text-uppercase"
              >
                Difícil
              </h5>
              <a
                className="Exercise__link text-uppercase"
                exact="true"
                href="/sexta-semana"
                onClick={[Function]}
              >
                Descubrir
              </a>
            </div>
            <div
              className="Exercise__border"
            />
          </article>
          <article
            className="Exercise"
          >
            <figure
              className="Exercise__imageContainer"
            >
              <img
                alt="Página para un portafolio"
                className="Exercise__image"
                loading="lazy"
                src="decima-tercera-semana-poster.jpg"
              />
            </figure>
            <div
              className="Exercise__details d-flex"
            >
              <h3
                className="Exercise__title text-uppercase"
              >
                Semana: 13
              </h3>
              <h5
                className="Exercise__difficulty text-uppercase"
              >
                Difícil
              </h5>
              <a
                className="Exercise__link text-uppercase"
                exact="true"
                href="/decima-tercera-semana"
                onClick={[Function]}
              >
                Descubrir
              </a>
            </div>
            <div
              className="Exercise__border"
            />
          </article>
          <article
            className="Exercise"
          >
            <figure
              className="Exercise__imageContainer"
            >
              <img
                alt="Landing para Fylo"
                className="Exercise__image"
                loading="lazy"
                src="decima-cuarta-semana-poster.jpg"
              />
            </figure>
            <div
              className="Exercise__details d-flex"
            >
              <h3
                className="Exercise__title text-uppercase"
              >
                Semana: 14
              </h3>
              <h5
                className="Exercise__difficulty text-uppercase"
              >
                Normal
              </h5>
              <a
                className="Exercise__link text-uppercase"
                exact="true"
                href="/decima-cuarta-semana"
                onClick={[Function]}
              >
                Descubrir
              </a>
            </div>
            <div
              className="Exercise__border"
            />
          </article>
        </div>
      </section>
    </div>
  </div>
</main>
`;
